[ADD] <sale_mrp_project_link> and <purchase_mrp_project_link>

agaldona opened the pull request.

Commit:
[ADD] sale_mrp_project

oihane commented on sale_mrp_project/views/project_project_view.xml:
```diff
@@ -0,0 +1,43 @@
+    </record>
+
+    </data>
+</openerp>
```
línea al final
![fireshot capture - add sale_mrp_project b_ - https___github com_odoomrp_odoomrp-wip_pull_252_files](https://cloud.githubusercontent.com/assets/2323616/4843399/9861ce56-6031-11e4-952e-5d93e5d0f1ea.png)

pedrobaeza commented:
El nombre del módulo no refleja lo que hace el módulo. Debería incluir lo de "general project" en algún lado, aunque "general project" tampoco me parece el mejor término. ¿Qué tal "main project"?

Además, hay que separar en sale_mrp_main_project, purchase_mrp_main_project, etc para cada una de las áreas, como @oihane ha hecho con otros módulos y activando el auto_install, para no perder la modularidad de si no se instala una de las partes.

agaldona commented:
Vale. Entonces, como veis que haga 3 módulos? 
- sale_main_project : este introduciria el campo de main project en sale y lo trasladaria a los procurements 
- mrp_main_project: cargaria desde los procurements a las OF-s y también a los nuevos procurements que se creen a partir de la OF
- purchase_main_project: este cargaría el main project en las compras que se crean desde los procurements
  No se si he llegado a explicarlo bien.

oihane commented:
Y cual sería el que lance la auto instalación?

agaldona commented:
Pues tanto mrp_main_project (dependiendo de sale_main_project y mrp) como purchase_main_project (dependiendo de sale_main_project y purchase). 
Creo.

pedrobaeza commented:
El enlace sería mrp_project_link o mrp_main_project. De hecho a lo mejor podrían ir juntos.

agaldona commented:
No entiendo.
Dices de unir mrp_project_link y mrp_main_project. Y que sale_main_project y purchase_main_project se carguen si sale y purchase están instalados además de mrp_main_project?

pedrobaeza commented:
A eso me refiero para la segunda parte. Para la primera, es ver si es lógico unirlos (que creo que sí, pero no lo conozco exactamente), es decir, ¿tiene sentido tener main_project en cuanto hacemos enlace con fabricación? Yo entiendo que sí. En ese caso, lo seguiríamos llamando al conjunto general _mrp_project_link_.

agaldona commented:
Si, al final el campo main_project de mrp seria el project_id que mete el mrp_project_link. El tema es, que el proceso de trasladarlo de procurements a las OF y al reves, es lo que hace la segunda parte. Que a su vez aprovecha para hacer ese mismo proceso tanto en ventas como en compras.
Yo no veo problema en ponerlos en uno, y añadir, dos módulos más. 
Solamente, como pongo los nombres? los unifico? o los dejo tal cual?

pedrobaeza commented:
El módulo principal: _mrp_project_link_. Y los otros dos: _sale_mrp_project_link_ y _purchase_mrp_project_link_. ¿Qué te parece?

agaldona commented:
Perfecto!

anajuaristi commented:
Yo no es por meterme en el tema... pero.. en este caso ¿para qué se necesita hacer esta separación y volver a tener que probar toda la funcionalidad de nuevo?
El módulo es de enlace entre proyectos, ventas, fabricación y compras y así está nombrado. ¿Para qué se requiere separar en partes si el módulo en sí define su enlace? No entiendo

Commit:
[DEL] se ha borrado el módulo sale_mrp_project
[ADD] nuevo módulo sale_mrp_project_link que depende del módulo mrp_project_link
[ADD] nuevo módulo purchase_mrp_project_link que depende del módulo mrp_project_link

agaldona commented:
Se ha eliminado sale_mrp_project y se han añadido los módulos sale_mrp_project_link y purchase_mrp_project_link. Estos dos módulos dependen den módulo mrp_project_link. #251

Commit:
[IMP] traducciones + flake8

agaldona commented:
Se me había colado pasar el Flake8 y añadir las traducciones.

pedrobaeza commented on purchase_mrp_project_link/models/project_project.py:
```diff
@@ -0,0 +1,35 @@
+        return result
+
+    purchase_count = fields.Integer(string='Purchase Count',
+                                    compute=_project_shortcut_count)
```
Si es un campo calculado, ¿por qué utilizas un método ya existente que no escribe ese dato? Deberías utilizar otro método.

agaldona replied:
Ah Ok! Pues básicamente ha sido porque lo he desglosado, a partir de un módulo donde se hacia todo en una función. No por otra cosa.

pedrobaeza commented on sale_mrp_project_link/models/project_project.py:
```diff
@@ -0,0 +1,35 @@
+        return result
+
+    sale_count = fields.Integer(string='Sale Count',
+                                compute=_project_shortcut_count)
```
Ídem que en compras.

Commit:
[IMP] modificaciones solicitadas por github

agaldona commented:
Cambios realizados.

pedrobaeza commented on sale_mrp_project_link/models/project_project.py:
```diff
@@ -0,0 +1,34 @@
+        sale_obj = self.env['sale.order']
+        sales = sale_obj.search([('main_project_id', '=', self.id)])
+        self.sale_count = len(sales)
+        return True
```
No hace falta devolver nada. Puedes quitarlo.

pedrobaeza commented on purchase_mrp_project_link/models/project_project.py:
```diff
@@ -0,0 +1,34 @@
+        purchase_obj = self.env['purchase.order']
+        purchases = purchase_obj.search([('main_project_id', '=', self.id)])
+        self.purchase_count = len(purchases)
+        return True
```
No hace falta devolver nada. Puedes quitarlo.

pedrobaeza commented:
Corrige esas dos últimas cosas, y cuando veas Travis verde, puedes mergear directamente.

Commit:
[IMP] quitar el return en las funciones de los campos calculados

agaldona commented:
Ok. Ya están los cambios.
